flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureManager.java

```diff
@@ -136,6 +135,8 @@ public void checkFailureCounter(CheckpointException exception, long checkpointId
                 // ignore
                 break;
 
+            case EXCEPTION:
```

Comment:
What is the reason for moving EXCEPTION out of ignore? I don't see any clue of this either in the ticket or in the tests.

Reply:
CheckpointFailureReason.EXCEPTION is also a kind of IOException may produced by StreamTaskNetworkInput.prepareSnapshot(), to keep the consistency, I think maybe we should moved it out of ignore. What do u think?

Reply:
I also doesn't see a reason why should we have both `EXCEPTION` and `IO_EXCEPTION`?  Unless we want to differentiate checkpoint failures on task vs job manager, but in that case names are miss-leading.

>  Why is EXCEPTION ignored in FailureManager right now?

I don't know. Maybe you can reach out to Arvid to ask about this, but I presume it's a simple omission.

Reply:
@AHeise In this PR new `CheckpointFailureReason` is introduced(`IO_EXCEPTION`). Now we want to replace the existing `EXCEPTION` with `IO_EXCEPTION`(because according to the code it is exactly IO_EXCEPTION). Also, `IO_EXCEPTION` would not be ignored by `FailureManager` as it was with `EXCEPTION`. Do you have any objections to this?

@zlzhang0122 If nobody will mind, I suggest getting rid of the EXCEPTION and replace it with IO_EXCEPTION. But it is better to do so in two separated commits(under one ticket). As a result of this ticket, you will have two commits - one of them adding the IO_EXCEPTION but don't touch the EXCEPTION and the second one just replace EXCEPTION with IO_EXCEPTION.

Reply:
Renaming should be completely fine as its not exposed or does it appear in the logs?
Do you plan to adjust the message? That could cause some issues if folks are parsing the logs, so maybe we can skip that.

Reply:
It, of course, appears in the logs somewhere(at least in stack trace but I don't see explicit printing of `CheckpointFailureReason`) but if we take into account that this PR split existing one `CheckpointFailureReason` into two which in any case changes the previous behavior I still think that it will be ok to rename it. And yes, the message will be slightly different.

Reply:
So you are not entirely replacing it, but only replacing it for IO related failures? Well then, there is nothing that you can do to maintain compability except making sure that the existing EXCEPTION is displaying as it was before. Make sure to point out the change in the release notes.

Reply:
Yes, we kind of want to replace only for IO-related failures but the thing is EXCEPTION is using only at one place and only for IO exception. So when we replace EXCEPTION with IO_EXCEPTION, there will be no other usages for EXCEPTION.

Reply:
Okay thanks for clarifying. In that case, do we really need to change the exception message? Could we maybe leave the original message and just add a small sentence that this most likely caused by a disk issue? Then old log monitoring will still work and you get even additional advise.